pkg/cloudprovider/providers/openstack/metrics/metrics.go

```diff
@@ -20,89 +20,57 @@ import (
 	"sync"
 	"time"
 
+	openstack_metrics "k8s.io/cloud-provider-openstack/pkg/metrics"
 	"k8s.io/component-base/metrics"
 	"k8s.io/component-base/metrics/legacyregistry"
 )
 
-type openstackMetrics struct {
-	duration *metrics.HistogramVec
-	total    *metrics.CounterVec
-	errors   *metrics.CounterVec
-}
-
 var (
-	reconcileMetrics = &openstackMetrics{
-		duration: metrics.NewHistogramVec(
+	occmReconcileMetrics = &openstack_metrics.OpenstackMetrics{
+		Duration: metrics.NewHistogramVec(
 			&metrics.HistogramOpts{
 				Name:    "cloudprovider_openstack_reconcile_duration_seconds",
 				Help:    "Time taken by various parts of OpenStack cloud controller manager reconciliation loops",
 				Buckets: []float64{0.01, 0.05, 0.1, 0.5, 1.0, 2.5, 5.0, 7.5, 10.0, 12.5, 15.0, 17.5, 20.0, 22.5, 25.0, 27.5, 30.0, 50.0, 75.0, 100.0, 1000.0},
 			}, []string{"operation"}),
-		total: metrics.NewCounterVec(
+		Total: metrics.NewCounterVec(
 			&metrics.CounterOpts{
 				Name: "cloudprovider_openstack_reconcile_total",
 				Help: "Total number of OpenStack cloud controller manager reconciliations",
 			}, []string{"operation"}),
-		errors: metrics.NewCounterVec(
+		Errors: metrics.NewCounterVec(
 			&metrics.CounterOpts{
 				Name: "cloudprovider_openstack_reconcile_errors_total",
 				Help: "Total number of OpenStack cloud controller manager reconciliation errors",
 			}, []string{"operation"}),
 	}
-	requestMetrics = &openstackMetrics{
-		duration: metrics.NewHistogramVec(
+	occmRequestMetrics = &openstack_metrics.OpenstackMetrics{
```

Comment:
@jichenjc `occmReconcileMetrics` are somehow specific to the CCM.  Whereas `occmRequestMetrics` is something that is reused by other parts of the OpenStack cloud provider, e.g. Cinder CSI Plugin.
Maybe it would be possible to move the `occmRequestMetrics` out of `pkg/cloudprovider/providers/openstack/metrics` to your proposed location `pkg/metrics/metrics.go`. What do you think?

Reply:
ok, that make sense, let me refactory the PR, thanks

Reply:
um... @seanschneeweiss another thought here , the definition created context
then it's up to OCCM or CSI cinder or others to create an instance for them
now occm created 2 , 1 for reconcile and the other for request metrics.. so keep occm_xxx
in existing places should be good? later on a new one named csi_cinder_xxx metrics will be created in cinder side ..
```
type MetricContext struct {
       Start            time.Time
       Attributes       []string
       ReconcileMetrics *OpenstackMetrics
       RequestMetrics   *OpenstackMetrics
}
```

Reply:
@jichenjc Please excuse my late respone. I would agree to define the metrics inside each component, and have the context creation inside each. That is somehow also the strategy other cloud providers use, imho. The OpenStack API metric (RequestMetrics) is slightly different, because it is used in so many places, not sure how to handle that. I'm fine with either way. But what we might have to keep in mind, that using the same metric names on multiple locations is not possible. So how to name the OpenStack API metrics within OCCM, how to name them within cinder csi? I expect to have a single name `openstack_api_request_...` for all API calls (no matter which component).

Reply:
um... so you mean , the calls either made by OCCM or CSI will be summed up and showed as one call, correct? 
ok, this is something different to my original thought , I thought we are running CSI/OCCM in different Pod and thus we are tracking there from different pod perspective, now I understand your point and need check it..

Reply:
@seanschneeweiss sorry, still learning here, let's say OCCM and CSI cinder both use openstack API 
is there someway that we can consolidate the API calls ?but they are in different container, can we do that?

Reply:
ok, let me try to move the reconcile logic back here and try again..

Reply:
@jichenjc looks good to me now, thank you very much.